dp-game-webapp/app/game/simulate.tsx

```diff
@@ -20,47 +20,71 @@ export function simulateConversions(
     seed: seed,
   });
 
-  const conversionCounts: number[] = [];
-
-  for (let i = 0; i < rounds; i++) {
-    conversionCounts.push(rand(impressions, conversionRates[i]));
+  for (const conversionRate of conversionRates) {
```

Comment:
so we do a binary distribution on mean and variance to get a value which we further use to do binomial distribution of conversion rates. n00b qn: why two steps?

Reply:
I need to write this up on the "About" page, but the basic idea is to think of the conversion rate like the weight of a coin, and the number of conversions like repeated tosses of that coin.

we assume different campaigns have different underlying conversion rates (typically measurement is an attempt to observe those differences.) so for each "result" we both sample a conversion rate from a Beta distribution, and then we sample the number of conversions given that specific conversion rate with a Binomial distribution.